metric_learn/_util.py

```diff
@@ -88,10 +87,6 @@ def check_input(input_data, y=None, preprocessor=None,
     is originally 1D and ``ensure_2d`` is True. Setting to 0 disables
     this check.
 
-  warn_on_dtype : boolean (default=False)

```

Comment:
scikit-learn 0.21 deprecates warn_on_dtype (see: https://github.com/scikit-learn/scikit-learn/issues/13324). Since we never use it (except to test it works), I suggest we get rid of it